lang/lua/lua.py

```diff
@@ -0,0 +1,308 @@
+ctx.matches = r"""
+tag: user.lua
+"""
+
+mod.setting(
+    "lua_version",
+    type=float,
+    default=5.1,
+    desc="The default lua version to use. Dictates certain operators",
+)
+mod.tag("stylua", desc="Tag for stylua linting commands")
+
+ctx.lists["user.code_common_function"] = {
+    "to number": "tonumber",
+    "I pairs": "ipairs",
+    "print": "print",
+    "print F": "printf",
+    "type": "type",
+    "assert": "assert",
+    "get meta table": "getmetatable",
+    "set meta table": "setmetatable",
+    # io
+    "I O write": "io.write",
+    "I O read": "io.read",
+    "I O open": "io.open",
+    # string
+    "format": "string.format",
+    "string G find": "string.gfind",
+    "string find": "string.strfind",
+    "string len": "string.strlen",
+    "string upper": "string.strupper",
+    "string lower": "string.strlower",
+    "string sub": "string.strsub",
+    "string G sub": "string.gsub",
+    "string match": "string.match",
+    "string G match": "string.gmatch",
+    # table
+    "table unpack": "table.unpack",
+    "table insert": "table.insert",
+    "tabel get N": "table.getn",
+    "tabel sort": "table.sort",
+    # math
+    "math max": "math.max",
+    # json
+    "jason parse": "json.parse",
+    # http
+    "H T T P get": "http.get",
+    "web get": "http.get",
+    # os
+    "O S date": "os.date",
+    "O S time": "os.time",
+    "O S clock": "os.clock",
+    "O S rename": "os.rename",
+    "O S remove": "os.remove",
+    "O S getenv": "os.getenv",
+    "O S execute": "os.execute",
+}
+
+ctx.lists["user.code_libraries"] = {
+    "bit": "bit",
+    "I O": "io",
+    "string": "string",
+    "U T F eight": "utf8",
+    "table": "table",
+    "math": "math",
+    "O S": "os",
+    "debug": "debug",
+    "L F S": "lfs",
+    "socket": "socket",
+    "H T T P": "http",
+    "web": "http",
+    "jason": "json",
+}
+
+
+@mod.capture(rule="{self.lua_functions}")
+def lua_functions(m) -> str:
+    "Returns a string"
+    return m.lua_functions
+
+
+@ctx.action_class("user")
+class UserActions:
+    # tag-related actions listed first, indicated by comment. corresponds to
+    # the tag(): user.code_imperative style declaration in the language .talon
+    # file
+
+    ##
+    # code_imperative
+    ##
+    def code_state_if():
+        actions.user.insert_between("if ", " then")
+
+    def code_state_else_if():
+        actions.user.insert_between("elseif ", " then")
+
+    def code_state_else():
+        actions.insert("else\n")
+
+    def code_state_do():
+        actions.insert("repeat\n")
+
+    def code_state_for():
+        actions.user.insert_between("for ", " do")
+
+    def code_state_go_to():
+        actions.insert("goto ")
+
+    def code_state_while():
+        actions.user.insert_between("while ", " do")
+
+    def code_state_return():
+        actions.insert("return ")
+
+    def code_break():
+        actions.insert("break ")
+
+    # Assumes a ::continue:: label
+    def code_state_continue():
```

Comment:
Is there a reason for defining a new action for this instead of reusing `code_state_next()`?

Reply:
Nope, just missed that code_next is what I needed when I implemented it I guess. Can change it to next instead.

Reply:
Probably appropriate. I had `state (next | continue): user.code_next()` in my local changes and ended up with the conflict.

Reply:
Sounds good. I'll have time tomorrow to fix it.